zeppelin-web/src/app/interpreter/interpreter.controller.js

```diff
@@ -723,16 +723,21 @@ function InterpreterCtrl ($rootScope, $scope, $http, baseUrlSrv, ngToast, $timeo
   }
 
   $scope.showSparkUI = function (settingId) {
-    $http.get(baseUrlSrv.getRestApiBase() + '/interpreter/getmetainfos/' + settingId + '?propName=url')
+    $http.get(baseUrlSrv.getRestApiBase() + '/interpreter/metadata/' + settingId)
       .success(function (data, status, headers, config) {
-        let url = data.body.url
-        if (!url) {
+        if (data.body === undefined) {
           BootstrapDialog.alert({
             message: 'No spark application running'
           })
           return
         }
-        window.open(url, '_blank')
+        if (data.body.url) {
+          window.open(data.body.url, '_blank')
+        } else {
+          BootstrapDialog.alert({
+            message: data.body.message
```

Comment:
I kept the same logic as before in the sense that, if Spark is not running, user gets popup: "No Spark application running". 
Cannot comment about a popup being disruptive or not. 
And on data.body.url being null and Spark running (only case we hit line 737 dialog), user gets either message "Spark UI disabled" which I think is correct or message "No spark url defined", which is a non-replicable scenario in a normal workflow.
I don't see anything we should change